R/nested.R

```diff
@@ -46,13 +49,17 @@ compute_parse_data_nested <- function(text) {
 #'   This is done recursively until we are only left with a nested tibble that
 #'   contains one row: The nested parse table.
 #' @param pd_flat A flat parse table including both terminals and non-terminals.
+#' @param top_level_expr The ids of the expressions that cannot be attributed
+#'   to another expression since they are not nested in any other expression.
+#'   This is important for terminating the recursion, i.e. further nesting is
+#'   impossible if our nested parse table has `length(top_level_expr)` rows.
 #' @seealso [compute_parse_data_nested()]
 #' @return A nested parse table.
-nest_parse_data <- function(pd_flat) {
-  if (nrow(pd_flat) <= 1) return(pd_flat)
+nest_parse_data <- function(pd_flat, top_level_expr) {
+  if (nrow(pd_flat) <= length(top_level_expr)) return(pd_flat)
```

Comment:
Yes please, this looks simpler to me than introducing a new argument to the recursive function. Be careful with operator precedence (`%in%` vs. `|`).